### ISSUE TITLE
Privacy feature: live stream opt-out

### ISSUE THREAD

**Issue by knah** [Yellow-Dog-Man/Resonite-Issues#1781]
### Is your feature request related to a problem? Please describe.

As a social platform, Resonite can enable people to connect to each other and share rather private things. At the same time, streaming social platforms is relatively popular. I personally have no desire to have my private conversations exposed to wide public, or to be used as content by some streamer rando. As such, an opt-out system for live streams would be nice to have.

### Describe the solution you'd like

An option in privacy settings for live stream opt-out. When enabled, users who are streaming (i.e. indicated by the "live" badge) will have both my avatar and voice hidden. I wouldn't mind a generic marker ("grey diamond") that indicates that I exist in a given spot to reduce social confusion about differentially visible users. Alternatively, the setting could be a three-state one: show/grey diamond/hide.
The guidelines might need to be updated to require streamers to use the "live" status in order to respect this opt-out on other users. A way to trigger it without using stream camera might be needed for that.

### Describe alternatives you've considered

Blocking streamers individually, which has quite a bit of downsides for this specific use case.
A mod that auto-blocks (and later unblocks) users with "live" indicator, I expect jankiness there, especially if I or the streamer happen to be the session host.
An avatar feature that replaces it with said grey diamond (and somehow kills the voice) if the observer is "live", but this should be a client feature, not a thing you have to do to each of your avatars.

### Additional Context

Vaguely inspired by [devlog discord post](https://discord.com/channels/1040316820650991766/1154514014563483759/1231004670122660002) showing screenshot opt-out

### Requesters

@͏knah

**Comment by AmasterAmaster** [comment]
I would not mind this as it is a pretty good idea, as the alternative I would also include is using a bit of flux to hide yourself automatically when a user is live in a world session, and also having a list of users in a bag/list that cant see you while they are streaming live. Then when they are done, the flux can detect that and reveal yourself once more to only that user. Granted, it is not intuitive for new users that dont know this or cant make said thing, so having a setting would be way better.

**Comment by epicEaston197** [comment]
I like this ideal but I'm concerned about this feature being confused with user blocking functionality as they are visually identical without contacts

**Comment by epicEaston197** [comment]
I like this ideal but I'm concerned about this feature being confused with user blocking functionality as they are visually identical without context

**Comment by JackTheFoxOtter** [comment]
I'm not sure I like this feature without the ability to overwrite it on a world level. When I host a live streaming world, and clearly mark it as such, I'd prefer people that come there actually be visible to the streaming camera. Also, when people don't use mirror to display, I'm not sure how you'd be able to only hide them from the screen recording, but not from the actual user view. Which would be a big self-moderation concern if I end up not seeing users while streaming either.

**Comment by knah** [comment]
@epicEaston197 hence the suggestion of a placeholder ("grey diamond") instead of complete invisibility, it can also include information on why the user is not visible.

@JackTheFoxOtter the ability to mark sessions as streamed can be tied into this. The scenario that causes me most annoyance is when streamers randomly join other users' sessions (i.e. contacts+/friends+ in VRC). It is easier to avoid sessions created by said streamers, especially if they were marked as such. This can also solve issues with moderation if the session host is streaming - privacy-conscious users can simply avoid marked sessions.

For users who use screen capture for their stream, I'd consider still requiring the same "live" status, either via manual toggle, or by detection of OBS or other common streaming/recording software. If it makes users invisible/replaced with placeholders in first-person view too, I'd say it's acceptable.

**Comment by kulzae** [comment]
Id probably rather see this as a world setting as its easier to moderate. asymmetrical allowance of recording is kind of wierd

Also id rather just have it completely remove any visual as If im trying to do a recording in a world im hosting id rather not have to deal with getting rid of floating grey rectangles in the background

**Comment by epicEaston197** [comment]
> Id probably rather see this as a session setting as its easier to moderate. asymmetrical allowance of recording is kind of wierd
> 
> Also id rather just have it completely remove any visual as If im trying to do a recording in a world im hosting id rather not have to deal with getting rid of floating grey rectangles in the background

I think the users should be only hidden to the camera similar to the photo capture privacy setting to minimize confusion
